src/operators.jl

```diff
@@ -66,3 +66,53 @@ end
 function Base.:(==)(a::SpinMonomial, b::SpinMonomial)
     return a.variables == b.variables
 end
+
+function Base.:(==)(var::SpinVariable, term::SpinTerm{T}) where T
+    # NOTE that 1*ax and ax are different types. One is SpinVariable{Int64}
+    # and the other is SpinVarible. Also note that ax*bx*bx=ax is of type
+    # SpinVariable{bool}.
+    if isone(coefficient(term))
+        return var == term.monomial
+    else
+        return false
+    end
+end
+
+function Base.:(==)(term::SpinTerm{T}, var::SpinVariable) where T
+    return var == term
+end
+
```

Comment:
Only this method is needed, the other ones added in this PR can be removed as they are equivalent to the default fallback implemented in MultivariatePolynomials.